src/backend/cpu/wrap.cpp

```diff
@@ -56,5 +56,45 @@ INSTANTIATE(uchar)
 INSTANTIATE(char)
 INSTANTIATE(short)
 INSTANTIATE(ushort)
+#undef INSTANTIATE
+
+template <typename T>
+Array<T> wrap_dilated(const Array<T> &in, const dim_t ox, const dim_t oy,
+                      const dim_t wx, const dim_t wy, const dim_t sx,
+                      const dim_t sy, const dim_t px, const dim_t py,
+                      const dim_t dx, const dim_t dy, const bool is_column) {
+    af::dim4 idims = in.dims();
+    af::dim4 odims(ox, oy, idims[2], idims[3]);
+
+    Array<T> out = createValueArray<T>(odims, scalar<T>(0));
+    out.eval();
+    in.eval();
+
+    const int d = is_column;
+    getQueue().enqueue(kernel::wrap_dim_dilated<T>, out, in, wx, wy, sx, sy, px,
+                       py, dx, dy, d);
+
+    return out;
+}
+
+#define INSTANTIATE(T)                                                      \
+    template Array<T> wrap_dilated<T>(                                      \
+        const Array<T> &in, const dim_t ox, const dim_t oy, const dim_t wx, \
+        const dim_t wy, const dim_t sx, const dim_t sy, const dim_t px,     \
+        const dim_t py, const dim_t dx, const dim_t dy, const bool is_column);
+
+INSTANTIATE(float)
+INSTANTIATE(double)
+INSTANTIATE(cfloat)
+INSTANTIATE(cdouble)
+INSTANTIATE(int)
+INSTANTIATE(uint)
+INSTANTIATE(intl)
+INSTANTIATE(uintl)
+INSTANTIATE(uchar)
+INSTANTIATE(char)
+INSTANTIATE(short)
+INSTANTIATE(ushort)
+#undef INSTANTIATE
```

Comment:
I think we can remove the instantiations for cfloat, cdouble , intl and uintl since I removed the those instantiations in strided convolution and convolve2Gradient @syurkevi Can you please confirm